### ISSUE TITLE
ur5 xarco not working?

### ISSUE THREAD

**Issue by b3l33** [ros-industrial/universal_robot#58]
Hello,

I am trying to generate 'ur5_robot.urdf' from xarco since the version of the 'ur5_robot.urdf' that is in the repo is not current (e.g., <plugin/> contains references to 'libgazebo_ros_controller_manager.so' when it should contain the updated reference to 'libgazebo_ros_control.so' in the latest version of 'gazebo.urdf.xacro'). When I run the following command:

>  rosrun xacro xacro.py ur5.urdf.xacro > ur5_robot.urdf

I get unexpected output contained in 'ur5_robot.urdf'.

![xarco](https://cloud.githubusercontent.com/assets/7157824/2682591/4b8cc2b2-c19d-11e3-93ec-15d36ab90afb.png)

Thanks in advance.

--Noob

**Comment by b3l33** [comment]
nevermind. found error....

**Comment by gavanderhoorn** [comment]
It would still be helpful if you could tell us what you did to get the correct output, if only to avoid having other users in the future reporting the same issue.

---

Also: which branch did you checkout?

**Comment by gavanderhoorn** [comment]
Also, @ipa-fxm, it would seem that the `.urdf` indeed are not up-to-date with the newer `xacros`. Should we address this?

**Comment by shaun-edwards** [comment]
I thought this was done here: 5c7464bd6e7514e209e35263ade5a104a6e4617c and tagged since 1.0.1.

**Comment by b3l33** [comment]
I was invoking xacro on the wrong file. I should have used 'ur5_robot.urdf.xacro':

> rosrun xacro xacro.py ur5_robot.urdf.xacro > ur5_robot.urdf

I downloaded the zip on 3/11/14. I was intending to acquire 'hydro' but I'm not sure if I got 'hydro-devel' instead.

Thank you for your responses.

**Comment by fmessmer** [comment]
Hmm, I don't understand why the content of the old version of gazebo.urdf.xacro is included within the URDFs and not the latest content...the URDFs have been generated later...maybe something got messed up because I had many different branches of universal_robot at that time.

@b3l33 Can you contribute back the correct URDFs in a PullRequest?

By the way, may I ask why you want to use the URDFs rather than the XACROs?

**Comment by b3l33** [comment]
I am new to ROS/Gazebo/etc. and as part of the learning process, I am trying to use URDFs directly to follow tutorials, etc. I am aware of how one would typically use XARCOs vs URDFs and will eventually use per best practice.

I may contribute back when I am past new issues that I have discovered. Prior to that I would want to discuss if we should instead remove the generated URDF and perhaps replace it with a README file that describes how to generate.

The new issue is that when I attempt to spawn the generated URDF Gazebo, Gazebo hangs and I see an INFO trace indicating that the plugin is waiting on the param server. I don't believe I saw this last week as I recall it working previously. So I'm not sure what has changed since I last reboot. Any suggestions? Should I open/reopen the issue? Thanks in advance. Examples follow...

> :~/catkin_ws/src/universal_robot/ur_description/urdf$ rosrun gazebo_ros spawn_model -file /home/user/catkin_ws/src/universal_robot/ur_description/urdf/ur5_robot.urdf -urdf -z 1 -model ur5

Console trace: 

[ INFO] [1397671984.537825418, 611.414000000]: gazebo_ros_control plugin is waiting for model URDF in parameter [robot_description] on the ROS param server.

**Comment by gavanderhoorn** [comment]
@b3l33: please open a new issue for this.

re: remove urdfs: (one of) the reason(s) for including the urdfs is because there were (and probably still are) some tools in ROS that we use, that are unable to load xacros directly. In order to avoid users to have to generate those files themselves, we included the urdfs (an example of such a tool was/is the MoveIt Setup assistant).

Also if our users install the binary distributions of packages, they have nowhere to store any generated urdfs, as the `/opt/ros/..` path is not user writeable.

**Comment by b3l33** [comment]
One more question on this issue... I noticed that the plugin name in 'gazebo.urdf.xacro' and subsequently the generated file 'ur5_robot.urdf' say "ros_control". Should this be "gazebo_ros_control" instead? It doesn't seem to matter - launching 'ur_gazebo/launch/ur5.launch' appears to work correctly. 

```
<plugin name="ros_control" filename="libgazebo_ros_control.so">
      <!--robotNamespace>/</robotNamespace-->
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
    </plugin>
```
